lib/cache/PackFileCacheStrategy.js

```diff
@@ -587,38 +626,39 @@ class PackContent {
 
 class PackFileCacheStrategy {
 	/**
-	 * @param {Object} options options
-	 * @param {IntermediateFileSystem} options.fs the filesystem
-	 * @param {string} options.context the context directory
-	 * @param {string} options.cacheLocation the location of the cache data
-	 * @param {string} options.version version identifier
-	 * @param {Logger} options.logger a logger
-	 * @param {Iterable<string>} options.managedPaths paths managed only by package manager
-	 * @param {Iterable<string>} options.immutablePaths immutable paths
+	 * @param {CacheStrategyOptions} options options
 	 */
 	constructor({
 		fs,
 		context,
+		contextify,
 		cacheLocation,
 		version,
 		logger,
 		managedPaths,
 		immutablePaths
 	}) {
-		this.fileSerializer = createFileSerializer(fs);
+		const contextifyMethods = createAbsolutePathsWriteReadMethods(
+			cacheLocation,
+			contextify
+		);
+		this.fileSerializer = createFileSerializer(fs, contextifyMethods);
 		this.fileSystemInfo = new FileSystemInfo(fs, {
 			managedPaths,
 			immutablePaths,
 			logger: logger.getChildLogger("webpack.FileSystemInfo")
 		});
+		this.contextifyAbsolutePath = contextifyMethods.writeAbsolutePath;
 		this.context = context;
+		this.contextify = contextify;
+		this.identCache = new Map();
```

Comment:
should be unneeded as `contextify` and `absolutify` already have caching. Extra logic should no longer be needed.